### ISSUE TITLE
Some output in command consoels spelled with speak typed characters enabled

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#513]
**Reported by aleksey_s on 2009-12-17 16:39**
Sometimes, NVDA is too verbose in console windows.
STR:
- open cmd
- run python interpreter
- type: print "hi"
  Currently, NVDA says
  h
  i
  hi
  It starts reading by letter and only then reads the entire message. This behaviour also exists in other console apps.
  **Blocked by #5027**
  **Blocking #2977, #4127, #5328**

**Comment by Adriani90** [comment]
I cannot reproduce this with NVDA 2018.2.1 on Windows 10.

**Comment by Adriani90** [comment]
@aleksey_s is this still an issue for you?

**Comment by leonardder** [comment]
I've seen it happening with Windows 10 as well

**Comment by codeofdusk** [comment]
Once #9771 is merged, this issue will be resolved on Windows 10 version 1803 and later due to the new `typedCharacter` implementation for UIA consoles. However, it is still not fixed on Windows Server and LTSC 2016 (Windows 10 1607), which will still be supported for many years to come. I've considered using `keyboardHandler` for legacy consoles (similar to UIA), but it is unsupported on 1607. I've also considered using `NVDAHelper` to receive wm_char messages, but they are unreliable (characters are sometimes doubled). Therefore, it seems that we cannot fix this bug on older Windows releases. Since the issue has either been resolved or resolution is impossible, I vote to close it. You are, of course, welcome to investigate further for legacy consoles and propose a solution if one exists!

**Comment by Adriani90** [comment]
What about People working on Windows 8.1 and Windows 7 in Consoles not using UIA?

**Comment by codeofdusk** [comment]
> What about People working on Windows 8.1 and Windows 7 in Consoles not using UIA?

The above comment is outdated, see #9915. This is fixed on a few Windows versions without UIA, but any more isn't possible due to doubled wm_char messages.
